### ISSUE TITLE
CLI --config option with shareable config in npm package no longer works

### ISSUE THREAD

**Issue by jharris4** [eslint/eslint#12023]
I tried to use an eslint config bundled in an npm package as described in https://github.com/eslint/eslint/issues/2543 and https://github.com/eslint/eslint/pull/3063 but it no longer seems to work in the latest versions of eslint.

Was this feature deliberately removed?

**Tell us about your environment**

* **ESLint Version:** 6.1.0
* **Node Version:** 12.4.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

N/A my question is about passing the config from CLI

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

N/A my question is about passing the config from CLI

```bash
eslint --config @my-scope/my-package
```

**What did you expect to happen?**

The config should have been loaded from the specified package

**What actually happened? Please include the actual, raw output from ESLint.**

`Error: Cannot read config file: /path/to/project/@my-scope/my-package`

**Are you willing to submit a pull request to fix this bug?**

Yes, if someone can help explain why this feature no longer works :-)

**Comment by platinumazure** [comment]
Hi @jharris4, thanks for the issue.

I can't think of any reason why the feature itself would have been changed or broken, but ESLint 6.0.0 did force resolution of configs to be local. Maybe there is a bug related to that. (In particular, the path noted in the error message is inconsistent- rather than looking in node_modules, it seems to be treating the specified config as a path unconditionally.)
